test/utils/cryptography/EIP712.test.js

```diff
@@ -21,12 +22,20 @@ contract('EIP712', function (accounts) {
       chainId: await getChainId(),
       verifyingContract: this.eip712.address,
     };
+    this.domainType = domainType(this.domain);
   });
 
-  it('domain separator', async function () {
-    const expected = await domainSeparator(this.domain);
+  describe('domain separator', function () {
+    it('is internally available', async function () {
+      const expected = await domainSeparator(this.domain);
 
-    expect(await this.eip712.$_domainSeparatorV4()).to.equal(expected);
+      expect(await this.eip712.$_domainSeparatorV4()).to.equal(expected);
+    });
+
+    it("can be rebuilt using EIP-5267's eip712Domain", async function () {
+      const rebuildDomain = await getDomain(this.eip712);
+      expect(mapValues(rebuildDomain, String)).to.be.deep.equal(mapValues(this.domain, String));
```

Comment:
I'm not very confident in this `mapValues(.., String)` operation. I wouldn't be surprised if some part of the domain is converted into a `"[object Object]"` string or something similar.

Reply:
would you prefer `mapValues(.., x => x.toString())` ???

Reply:
I checked and `String` seems to be working ok. Let's leave it.